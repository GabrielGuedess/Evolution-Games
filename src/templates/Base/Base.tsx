import { Footer } from 'components/organisms/Footer/Footer';
import { Navbar } from 'components/organisms/Navbar/Navbar';

import * as S from './Base.styles';

export type BaseTemplateProps = {
  children: React.ReactNode;
  hasColors?: boolean;
  hasMarginTop?: boolean;
};

export const Base = ({
  children,
  hasColors = false,
  hasMarginTop = false,
}: BaseTemplateProps) => (
  <S.Wrapper>
    {hasColors && (
      <>
        <S.CircleBlurLeft aria-label="Circle left blur" />
        <S.CircleBlurRight aria-label="Circle right blur" />
      </>
    )}

    <Navbar />

    <S.Content aria-label="Content" hasMarginTop={hasMarginTop}>
      {children}
    </S.Content>

    <S.SectionFooter>
      <Footer />
    </S.SectionFooter>
  </S.Wrapper>
);
