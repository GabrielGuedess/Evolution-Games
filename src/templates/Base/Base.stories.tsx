import { Story, Meta } from '@storybook/react/types-6-0';

import { Container } from 'components/atoms/Container/Container';

import { Base, BaseTemplateProps } from './Base';

export default {
  title: 'Templates/Base',
  component: Base,
  parameters: {
    layout: 'fullscreen',
  },
  args: {
    hasColors: true,
    hasMarginTop: true,
  },
} as Meta<BaseTemplateProps>;

export const Default: Story<BaseTemplateProps> = args => (
  <Base {...args}>
    <Container>
      <h1>Base</h1>
    </Container>
  </Base>
);
