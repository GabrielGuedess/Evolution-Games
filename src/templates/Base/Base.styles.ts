import media from 'styled-media-query';

import styled, { css } from 'styled-components';

import { BaseTemplateProps } from './Base';

export const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  max-width: 100vw;
  overflow: hidden;
  justify-content: space-between;
  z-index: 10;
`;

export const Content = styled.div<Pick<BaseTemplateProps, 'hasMarginTop'>>`
  ${({ hasMarginTop }) => css`
    margin-top: ${hasMarginTop ? '7.8rem' : 0};
    flex: 1 0 auto;

    ${media.greaterThan('medium')`
      margin-top: ${hasMarginTop ? '10.4rem' : 0};
    `}
  `}
`;

export const SectionFooter = styled.div`
  margin-top: 6rem;
`;

export const CircleBlurLeft = styled.div`
  position: absolute;
  inset: 0 auto auto 0;
  transform: translate(-50%, -50%);
  width: 55.2rem;
  aspect-ratio: 1/1;
  z-index: -1;
  border-radius: 50%;
  filter: blur(12.5rem);
  background: linear-gradient(
    180deg,
    rgba(118, 74, 241, 0.8),
    rgba(118, 74, 241, 0)
  );
`;

export const CircleBlurRight = styled.div`
  position: absolute;
  inset: 0 0 auto auto;
  transform: translate(50%, -50%);
  width: 55.2rem;
  aspect-ratio: 1/1;
  z-index: -1;
  border-radius: 50%;
  filter: blur(12.5rem);
  background: linear-gradient(
    180deg,
    rgba(118, 74, 241, 0.8),
    rgba(118, 74, 241, 0)
  );
`;
