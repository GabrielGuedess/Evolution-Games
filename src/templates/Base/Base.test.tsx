import { render, screen } from '@testing-library/react';

import { Base } from './Base';

jest.mock('components/organisms/Navbar/Navbar', () => ({
  __esModule: true,
  Navbar: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Navbar Mock">{children}</div>;
  },
}));

jest.mock('components/organisms/Footer/Footer', () => ({
  __esModule: true,
  Footer: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Footer Mock">{children}</div>;
  },
}));

describe('<Base />', () => {
  it('should render component correctly', () => {
    // Arrange
    const { container } = render(<Base>Test</Base>);

    // Assert
    expect(screen.getByTestId('Navbar Mock')).toBeInTheDocument();
    expect(screen.getByTestId('Footer Mock')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render component with margin', () => {
    render(<Base hasMarginTop>Test</Base>);

    // Assert
    expect(screen.getByLabelText('Content')).toHaveStyle({
      'margin-top': '7.8rem',
    });
  });

  it('should render component with circle blur', () => {
    render(<Base hasColors>Test</Base>);

    // Assert
    expect(screen.getByLabelText('Circle left blur')).toBeInTheDocument();
    expect(screen.getByLabelText('Circle right blur')).toBeInTheDocument();
  });
});
