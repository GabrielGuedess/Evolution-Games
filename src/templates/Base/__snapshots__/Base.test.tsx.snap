// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Base /> should render component correctly 1'] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
  max-width: 100vw;
  overflow: hidden;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  margin-top: 0;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c2 {
  margin-top: 6rem;
}

@media (min-width:768px) {
  .c1 {
    margin-top: 0;
  }
}

<div
  class="c0"
>
  <div
    data-testid="Navbar Mock"
  />
  <div
    aria-label="Content"
    class="c1"
  >
    Test
  </div>
  <div
    class="c2"
  >
    <div
      data-testid="Footer Mock"
    />
  </div>
</div>
`;
